kotlintest-runner/kotlintest-runner-jvm/src/main/kotlin/io/kotlintest/runner/jvm/TestCaseExecutor.kt

```diff
@@ -163,25 +167,25 @@ class TestCaseExecutor(private val listener: TestEngineListener,
   private fun before(testCase: TestCase) {
     listener.enterTestCase(testCase)
 
-    val userListeners = testCase.spec.listeners() + Project.listeners()
-    userListeners.forEach { it.beforeTest(testCase.description) }
-
-    if (testCase.config.enabled) {
-      // Only execute before test from the spec if the test is enabled
-      testCase.spec.beforeTest(testCase.description)
+    val userListeners = testCase.spec.listeners() + testCase.spec + Project.listeners()
+    userListeners.forEach {
+      it.beforeTest(testCase.description)
+      if (isActive(testCase)) {
+        it.beforeTest(testCase)
```

Comment:
Wouldn't there be scenarios in which we still want to listen to a test even if it's inactive? For instance, if I want to override the testcase execution.

I set it to only ignore if it was a `spec` because by default we don't want to execute an ignored testcase when we're on a spec.

I think that `isActive` should only be verified if we're running from a spec.
Also, by executing the deprecated method before we verify if the test is active, we aren't allowing some of our own tests to fail if they should, as the scenario would be hidden inside this if

Reply:
My first point is: We should never execute `beforeTest` if the test is inactive, unless an Extension of some sort wants to do it. There is a test (I don't remember which, but it broke recently) that exercises this behavior.

My second point, and the last paragraph: In line 172, we're using a deprecated method, right? If we do that for deprecations, some of our tests won't test the new behavior, they'll keep testing the old one